interface/main/tabs/main.php

```diff
@@ -222,13 +221,39 @@ function jsFetchGlobals(scope) {
         })
     })
 }
-
 jsFetchGlobals('top').then(globalJson => {
     jsGlobals = globalJson;
 }).catch(error => {
     console.log(error.message);
 });
 
+// set up global translations for js
+function setupI18n(lang_id) {
+    top.restoreSession();
+    return new Promise((resolve, reject) => {
```

Comment:
fetch already returns a promise so making the function call a promise is moot. Simply return fetch e.g return fetch(...)
Remove the Promise wrapper.
fyi, fetch won't reject http errors you have to test for them i.e response.ok else Promise.reject('error') ....